react-native/typescript/AmpliApp/middleware/segmentMiddleware.ts

```diff
@@ -0,0 +1,61 @@
+import {createClient, Config, JsonMap} from '@segment/analytics-react-native';
+import {
+  BaseEvent,
+  IdentifyEvent,
+  Middleware,
+  SpecialEventType,
+} from '@amplitude/react-native';
+
+export type SegmentExtra = {
+  segment?: {
+    userId?: string;
+  };
+};
+
+/**
+ * getSegmentMiddleware
+ *
+ * Initializes Segment tracker and returns Segment Middleware
+ */
+export function getSegmentMiddleware(config: Config): Middleware {
+  // Create Segment tracker
+  const analytics = createClient(config);
+
+  // Create Segment Middleware
+  return (payload, next) => {
+    const {event, extra} = payload;
+    const segmentExtra = extra as SegmentExtra;
+    const userId = segmentExtra?.segment?.userId;
```

Comment:
I think `userId` might be confusing here but don't see `anonymousId` in the Segment RN SDK. Can we use another segment specific option like `integrations`? https://segment.com/docs/connections/sources/catalog/libraries/mobile/react-native/#block-specific-events-from-going-to-a-destination

Reply:
It looks the current version of Segment RN SDK (2+) doesn't support any additional track options:
```
track(eventName: string, options?: JsonMap)
```
https://github.com/segmentio/analytics-react-native/blob/master/packages/core/src/analytics.ts#L484

I use `2.1.7` version - maybe it is beta?
```
"@segment/analytics-react-native": "^2.1.7"
```

Should I use stable `1.x.y` version instead?

BTW, I changed `userId` logic

Reply:
Current implementation with 2.1.7 looks good. Thanks!